src/renderer/qssWidget/js/qssVolumeWidget.js

```diff
@@ -0,0 +1,495 @@
+                extendedTip: "{that}.model.setting.widget.extendedTip"
+            }
+        },
+        events: {
+            onNotificationRequired: null
+        },
+        listeners: {
+            "onCreate": {
+                this: "{that}.dom.helpImage",
+                method: "attr",
+                args: ["src", "{that}.model.setting.schema.helpImage"]
+            }
+        },
+        modelListeners: {
+            value: {
+                funcName: "gpii.qssWidget.volume.updateSwitchState",
+                args: ["{switchButton}", "{stepper}", "{change}.value"]
+            }
+        },
+
+        components: {
+            stepper: {
+                type: "gpii.qssWidget.volumeStepper",
+                container: "{that}.dom.stepper",
+                options: {
+                    model: {
+                        setting: "{volume}.model.setting",
+                        value: "{volume}.model.value",
+                        messages: {
+                            on: null,
+                            off: null
+                        }
+                    },
+                    sounds: "{volume}.options.sounds",
+                    events: {
+                        onNotificationRequired: "{volume}.events.onNotificationRequired"
+                    }
+                }
+            },
+            switchButton: {
+                type: "gpii.psp.widgets.switch",
+                container: "{that}.dom.switch",
+                options: {
+                    model: {
+                        enabled: {
+                            expander: {
+                                funcName: "gpii.qssWidget.volume.transformValue",
+                                args: ["{volume}.model.setting.value"]
+                            }
+                        },
+                        messages: {
+                            on: "{volume}.model.messages.on",
+                            off: "{volume}.model.messages.off"
+                        }
+                    },
+                    invokers: {
+                        toggleModel: {
+                            funcName: "gpii.qssWidget.volume.toggleModel",
+                            args: ["{that}", "{volume}", "{stepper}", "{channelNotifier}.events.onQssWidgetSettingAltered"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    /**
+     * Invoked whenever the volume value is changed and updating the state of the
+     * volume switch button.
+     * @param {Component} switchButton - The `gpii.psp.widgets.volume.switchButton` instance.
+     * @param {Component} stepper - The `gpii.psp.widgets.volumeStepper instance.
+     * @param {Number} value - The value of the setting.
+     */
+    gpii.qssWidget.volume.updateSwitchState = function (switchButton, stepper, value) {
+        if (value === 0 && !switchButton.model.enabled) {
```

Comment:
This can be more clearly written
if (!value !== switchButton.model.enabled) {
    switchButton.applier.change("enabled", !switchButton.model.enabled, null, "settingAlter")
}

Reply:
Done, take a look:
https://github.com/GPII/gpii-app/pull/136